lib/DxcSupport/WinAdapter.cpp

```diff
@@ -75,4 +77,49 @@ CHandle::CHandle(HANDLE h) { m_h = h; }
 CHandle::~CHandle() { CloseHandle(m_h); }
 CHandle::operator HANDLE() const throw() { return m_h; }
 
+// CComBSTR
+CComBSTR::CComBSTR(_In_ int nSize, LPCWSTR sz) : m_needFree(true) {
+  if (nSize < 0) {
+    throw  std::invalid_argument("CComBSTR must have size >= 0");
+  }
+
+  if (nSize == 0) {
+    m_str = NULL;
+  } else {
+    m_str = (BSTR)malloc((nSize + 1) * sizeof(WCHAR));
```

Comment:
This does not look correct to me.  A BSTR is supposed to start with a 4-byte prefix for the size of the string, then the string pointer points to the first character of the string, after the prefix.

Reply:
See:
https://learn.microsoft.com/en-us/previous-versions/windows/desktop/automat/bstr

Reply:
Switch to SysAllocStringLen.